experimental/dds/tree2/src/feature-libraries/sequence-field/sequenceFieldChangeEncoder.ts

```diff
@@ -20,110 +20,65 @@ function makeV0Codec<TNodeChange>(
 		encode: (changeset) => {
 			const jsonMarks: JsonCompatible[] = [];
 			for (const mark of changeset) {
+				// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+				const encodedMark = { ...mark } as Mark<JsonCompatibleReadOnly>;
+				if (mark.changes !== undefined) {
+					encodedMark.changes = childCodec.encode(mark.changes);
+				}
 				const type = mark.type;
 				switch (type) {
+					case NoopMarkType:
+					case "MoveIn":
+					case "ReturnTo":
 					case "Insert":
 					case "Delete":
 					case "MoveOut":
 					case "ReturnFrom":
-						if (mark.changes !== undefined) {
-							jsonMarks.push({
-								...mark,
-								changes: childCodec.encode(mark.changes),
-							} as unknown as JsonCompatible);
-						} else {
-							jsonMarks.push(mark as Mark<TNodeChange> & JsonCompatible);
-						}
-
 						break;
 					case "Revive": {
-						const content = mark.content.map(jsonableTreeFromCursor);
-						if (mark.changes !== undefined) {
-							jsonMarks.push({
-								...mark,
-								content,
-								changes: childCodec.encode(mark.changes),
-							} as unknown as JsonCompatible);
-						} else {
-							jsonMarks.push({
-								...mark,
-								content,
-							} as unknown as JsonCompatible);
-						}
+						(encodedMark as unknown as { content: JsonCompatible[] }).content =
+							mark.content.map(jsonableTreeFromCursor) as unknown as JsonCompatible[];
```

Comment:
I'm not sure this is any worse than it was before, but I do wonder if there's a better way to do this.

Reply:
I can't begin to express my frustration and disgust at having to do this. We have a work item to decouple the in-memory representation from the over-the-wire one, and that will make things better. I just figured it's better to wait until our in-memory format settles down some more, so we're not making more refactoring work for outselves.